contracts/utility/TokenForwarding.cdc

```diff
@@ -43,6 +48,14 @@ pub contract TokenForwarding {
             emit ForwardedDeposit(amount: balance, from: self.owner?.address)
         }
 
+        pub fun check(): Bool {
+            return self.recipient.check<&{FungibleToken.Receiver}>()
+        }
+
+        pub fun safeBorrow(): &{FungibleToken.Receiver}? {
+            return self.recipient.borrow<&{FungibleToken.Receiver}>()
+        }
```

Comment:
Thinking through this.. Is there a scenario where this borrow panics? Isn't that what happens after secure cadence if the completely wrong type is underneath?

In that case, we will need to check first, and then return the borrow result afterwards

Reply:
that is a good point. I think you might be right. better to check beforehand